### PR TITLE
Fuel Grid Changes

### PR DESCRIPTION
**Pull request by brettedw**
- Removes raster reprojections from HFI calcs
- Reprojects polygons from database to match raster
- Goal is to reduce fuel grid file dependencies
   - Going forward remove need for `fuel_types_from_sfms_epsg_3005.tif` and `fuel_types_from_sfms_epsg_3005.shp` (though the shp is currently still used in past migrations)
# Test Links:
[Landing Page](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3637-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Review by conbrad — APPROVED** [review]
Nice work, that simplifies things a lot!

File: api/app/auto_spatial_advisory/process_fuel_type_area.py [hunk 1]
```diff
@@ -112,11 +111,63 @@ def intersect_raster_by_advisory_shape(threshold: int, advisory_shape_id: int, s
...
+
+    for feature in layer:
+        geometry = feature.GetGeometryRef()
+        geometry.Transform(coordinate_transform)
+        new_feature = ogr.Feature(reprojected_layer.GetLayerDefn())
+        new_feature.SetGeometry(geometry)
+        reprojected_layer.CreateFeature(new_feature)
+        new_feature = None
+
+    return output_path
```
Comment (dgboss) [inline api/app/auto_spatial_advisory/process_fuel_type_area.py:168]: Should we be setting reprojected_ds to None before returning? I've read mixed reviews on allowing garbage collection to handle data source clean up versus explicitly closing the references.

Comment (dgboss) [reply api/app/auto_spatial_advisory/process_fuel_type_area.py:168]: Also not sure if layers should be set to None or not?!

**Review by dgboss — APPROVED** [review]
Nice work! Just one question about manually closing a data source.

File: api/app/auto_spatial_advisory/process_fuel_type_area.py (hunk 1 above)
Comment (brettedw) [reply api/app/auto_spatial_advisory/process_fuel_type_area.py:168]: Good questions. I've read once the objects are out of scope the garbage collector should [handle it](https://gdal.org/api/python_gotchas.html#:~:text=This%20is%20typically%20unnecessary%20because%20these%20objects%20are%20automatically%20destroyed%20during%20garbage%20collection%20when%20no%20references%20to%20the%20Python%20object%20remain.). But it also seems it's a good idea to clean up datasource objects explicitly according to GDAL at the link above.

On another note, I saw while doing this that in GDAL 3.8 context managers are available.
`with ogr.Open() as.....`

Comment (conbrad) [reply api/app/auto_spatial_advisory/process_fuel_type_area.py:168]: I believe our base image is running a super old GDAL version (3.3 or 3.4?) so we should consider rebuilding it soon.

**Pull request merged**